### ISSUE TITLE
Upgrade to slf4j 2.0.1 breaks all log4j logging

### ISSUE THREAD

**Issue by quotidian-ennui** [adaptris/interlok-build-parent#130]
- Checkout [json-csv](https://github.com/adaptris-labs/build-parent-json-csv)
- Edit the build.gradle so it refers to the develop branch (change main -> develop)
- `gradle assemble`
- Start Interlok

```
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions 1.7.x or earlier.
SLF4J: Ignoring binding found at [jar:file:/C:/storage/work/runtime/interlok/json-csv/build/distribution/lib/log4j-slf4j-impl.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See https://www.slf4j.org/codes.html#ignoredBindings for an explanation.
Bootstrap of Interlok 4.5.0-RELEASE(2022-06-22:09:12:12 UTC) complete
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.thoughtworks.xstream.converters.reflection.FieldDictionary (file:/C:/storage/work/runtime/interlok/json-csv/build/distribution/lib/xstream.jar) to field java.util.AbstractCollection.MAX_ARRAY_SIZE
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.converters.reflection.FieldDictionary
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Running ShutdownHandler, please wait...
MLog initialization issue: slf4j found no binding or threatened to use its (dangerously silent) NOPLogger. We consider the slf4j library not found.
```

**Comment by sebastien-belin-adp** [comment]
I've done a fix in interlok-core today, that should fix it tomorrow hopefully

**Comment by quotidian-ennui** [comment]
The "fix" is a backwards step in behaviour; this [commit](https://github.com/adaptris/interlok/commit/da0dc549c76ee746ecd3c7546625fce562538eef) _works_ but is essentially delegating logging through 3 logging systems which is crazy just to upgrade something that purports to be fully backwardly compatible!

> Essentially with that commit we are now effectively doing SLF4J -> Log4j1.x -> Log4j2.x from that I understand of the various shims in place. This actually presents a problem because "order may well be important" because 

```
// The main API
interlokRuntime ("org.slf4j:slf4j-api:$slf4jVersion")
// Provides an SLF4J Adapter that pretends to be log4j1
interlokRuntime ("org.slf4j:slf4j-reload4j:$slf4jVersion")
// Provides the log4j2 shim for log4j1 logging.
interlokRuntime ("org.apache.logging.log4j:log4j-1.2-api:$log4j2Version")

// Does the actual logging via log4j2
interlokRuntime ("org.apache.logging.log4j:log4j-core:$log4j2Version")
interlokRuntime ("org.apache.logging.log4j:log4j-api:$log4j2Version")
```


We should actually be using `org.apache.logging.log4j:log4j-slf4j18-impl` however, this some exception logging on startup (even though logging still works).

Associated [log4j2-issue](https://issues.apache.org/jira/browse/LOG4J2-3139) - TLDR; will be targeted for a fix in log4j2-2.19.0

**Comment by quotidian-ennui** [comment]
Note that because slf4j-reload4j is marked as optional, fixes in interlok-core won't be propagated here, so you need to fix it here as well 

I'd actually suggest that you do this _until log4j2-2.19_ is released, and then test with log4j-sl4j18-impl (or whatever it will be called).

```
slf4jVersion = '1.7.36'
  interlokRuntime ("org.slf4j:slf4j-api") {
    version {
      strictly "$slf4jVersion"
      because "log4j-slf4j18-impl is not compatible with slf4j2.x"
    }
  }
  interlokRuntime ("org.slf4j:jcl-over-slf4j:$slf4jVersion")  {
    version {
      strictly "$slf4jVersion"
      because "log4j-slf4j18-impl is not compatible with slf4j2.x"
    }
  }
  interlokRuntime ("org.slf4j:jul-to-slf4j:$slf4jVersion") {
    version {
      strictly "$slf4jVersion"
      because "log4j-slf4j18-impl is not compatible with slf4j2.x"
    }
  }
```
